content/general/concepts/jit-compilation/jit-compilation.md

```diff
@@ -0,0 +1,41 @@
+---
+Title: 'Just-In-Time (JIT) Compilation'
+Description: 'Just-In-Time (JIT) compilation is a system for compiling languages as they are executed, rather than compiling prior to execution.'
+Subjects:
+  - 'Developer Tools'
+Tags:
+  - 'Developer Tools'
+  - 'Compilation'
+CatalogContent:
+  - 'learn-java'
+  - 'introduction-to-javascript'
+  - 'paths/computer-science'
+---
+
+Just-In-Time (JIT) compilation is a system for compiling languages as they are executed, rather than compiling prior to execution. In JIT compilation, the compiling happens during the [runtime](https://www.codecademy.com/resources/docs/general/runtime), which is a built-in application that supports the execution of a program. Runtime generally refers to the infrastructure required to implement features in the language itself.
+
+JIT compilation can be thought of as a combination of traditionally compiled languages and interpreted languages. Traditionally compiled languages are languages where the computer directly translates the code and program prior to execution. Interpreted languages are not translated by the computer, but instead, they use an interpreter to read and execute the code directly. [C](https://www.codecademy.com/resources/docs/c) is an example of a traditionally compiled language, and [Python](https://www.codecademy.com/resources/docs/python) is an example of an interpreted language. JIT compilation is a mixture of these two methods.
```

Comment:
Should mention what the compiler translates the code into. (With a link to the machine code entry.)